### PR TITLE
Fixing byte[] as groupby value

### PR DESCRIPTION
**Pull request by xiangfu0**
Fixing the group by query of bytes column by converting the bytes to hex string underlying.

### CODE REVIEW

File: pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/groupby/DictionaryBasedGroupKeyGenerator.java [hunk 1]
```diff
@@ -421,21 +422,28 @@ public void remove() {
...
         cardinality = _cardinalities[i];
-        groupKeyBuilder.append(_dictionaries[i].get(rawKey % cardinality));
+        groupKeyBuilder.append(getStringFromDictVal(_dictionaries[i].get(rawKey % cardinality)));
         rawKey /= cardinality;
       }
       return groupKeyBuilder.toString();
     }
   }
 
+  private String getStringFromDictVal(Object value) {
```
Comment (kishoreg) [inline pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/groupby/DictionaryBasedGroupKeyGenerator.java]: don't we have getString method in Dictionary?

Comment (kishoreg) [reply pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/groupby/DictionaryBasedGroupKeyGenerator.java]: we should restrict the Hex.encodeHexString to ByteDictionary if possible

Comment (xiangfu0) [reply pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/groupby/DictionaryBasedGroupKeyGenerator.java]: Good point, will override this method.

Comment (xiangfu0) [reply pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/groupby/DictionaryBasedGroupKeyGenerator.java]: Updated BytesDictionary's getString method and use it in the groupKeyGenerator

File: pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/groupby/DictionaryBasedGroupKeyGenerator.java [hunk 2]
```diff
@@ -28,6 +28,7 @@
 import java.util.Iterator;
 import java.util.NoSuchElementException;
 import javax.annotation.Nonnull;
+import org.apache.commons.codec.binary.Hex;
```
Comment (kishoreg) [inline pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/groupby/DictionaryBasedGroupKeyGenerator.java]: we dont need this anymore rt?

**Review by kishoreg — APPROVED** [review]
LGTM. let's wait for CI to pass

**Pull request merged**